config-model-api/src/main/java/com/yahoo/config/model/api/TenantSecretStore.java

```diff
@@ -30,10 +32,12 @@ public String getRole() {
         return role;
     }
 
-    public boolean isValid() {
-        return !name.isBlank() &&
-                !awsId.isBlank() &&
-                !role.isBlank();
+    public Optional<String> getExternalId() {
+        return externalId;
+    }
+
+    public void setExternalId(String externalId) {
```

Comment:
consider renaming to withExternalId and return a new TenantSecretStore. Make this class immutable.